### PR TITLE
Add order process list page

### PR DESCRIPTION
**Pull request by lgalis**
https://projects.engineering.redhat.com/browse/SSP-1709

Add a new page for the list of Order processes - includes sorting and pagination.
Add redux methods for order processes.

Screenshot from 2020-07-22 18-32-27

Screenshot from 2020-07-23 18-20-24

Screenshot from 2020-07-23 00-24-01

### CODE REVIEW

File: src/Routes.js [hunk 1]
```diff
@@ -48,10 +48,16 @@ const OrderDetail = lazy(() =>
     /* webpackChunkName: "order-detail" */ './smart-components/order/order-detail/order-detail'
   )
 );
+const OrderProcesses = lazy(() =>
+  import(
+    /* webpackChunkName: "platforms" */ './smart-components/order-process/order-processes'
```
Comment (Hyperkid123) [inline src/Routes.js:53]: the chunk name should be unique

File: src/messages/common.message.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    id: 'commonMessages.delete',
+    defaultMessage: 'Delete'
+  },
+  deleting: {
+    id: 'commonMessages.deleting',
+    defaultMessage: 'Deleting'
+  },
+  cancel: {
+    id: 'commonMessages.cancel',
+    defaultMessage: 'Cancel'
```
Comment (Hyperkid123) [inline src/messages/common.message.js:30]: There should already be afile with most of these messages defined.

File: src/messages/order-processes.messages.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    id: 'order-process.toolbar.order-process',
+    defaultMessage: 'Order process'
+  },
+  description: {
+    id: 'order-process.table.description',
+    defaultMessage: 'Description'
+  },
+  created_at: {
+    id: 'order-process.table.created_at',
+    defaultMessage: 'Created'
```
Comment (Hyperkid123) [inline src/messages/order-processes.messages.js:38]: This message is already defined at `states.messages.js`

File: src/messages/order-processes.messages.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    id: 'order-process.toolbar.order-processes',
+    defaultMessage: 'Order processes'
+  },
+  orderProcess: {
+    id: 'order-process.toolbar.order-process',
+    defaultMessage: 'Order process'
+  },
+  description: {
+    id: 'order-process.table.description',
+    defaultMessage: 'Description'
```
Comment (Hyperkid123) [inline src/messages/order-processes.messages.js:34]: This message is defined inside `labels.messages.js`

File: src/messages/table-empty.messages.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const { defineMessages } = require('react-intl');
+
+const tableEmptyMessages = defineMessages({
+  noRecords: {
+    id: 'tableEmpty.noRecords',
+    defaultMessage: 'No records'
```
Comment (Hyperkid123) [inline src/messages/table-empty.messages.js:6]: There are some filtering messages for empty states in this file `filtering.messages.js`. We should not introduce new strings. We have a lot of near-duplicates already.

File: src/messages/table-toolbar.messages.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    id: 'tableToolbar.noResultsFound',
+    defaultMessage: 'No results found'
+  },
+  noResult: {
+    id: 'tableToolbar.noResult',
+    defaultMessage: 'No {results}'
+  },
+  clearAllFilters: {
+    id: 'tableToolbar.clearAllFilters',
+    defaultMessage: 'Clear all filters'
```
Comment (Hyperkid123) [inline src/messages/table-toolbar.messages.js:22]: ^^

File: src/presentational-components/shared/table-toolbar-view.js [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+  defaultSettings,
+  getCurrentPage,
+  getNewPage
+} from '../../helpers/shared/pagination';
+import { ListLoader } from './loader-placeholders';
+import { useIntl } from 'react-intl';
+import { Section } from '@redhat-cloud-services/frontend-components/components/cjs/Section';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/cjs/PrimaryToolbar';
+
+export const TableToolbarView = ({
```
Comment (Hyperkid123) [inline src/presentational-components/shared/table-toolbar-view.js:14]: I believe we don't really need most of these props. Can you reduce the list a bit?

File: src/presentational-components/shared/table-toolbar-view.js [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+  createRows,
+  columns,
+  fetchData,
+  toolbarButtons,
+  data,
+  actionResolver,
+  actionsDisabled,
+  routes,
+  titlePlural,
+  titleSingular,
```
Comment (Hyperkid123) [inline src/presentational-components/shared/table-toolbar-view.js:25]: These should be handled by the translation API.

File: src/smart-components/order-process/order-processes.js [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+  const intl = useIntl();
+
+  const updateOrderProcesses = (pagination) => {
+    stateDispatch({ type: 'setFetching', payload: true });
+    return dispatch(fetchOrderProcesses(pagination))
+      .then(() => stateDispatch({ type: 'setFetching', payload: false }))
+      .catch(() => stateDispatch({ type: 'setFetching', payload: false }));
+  };
+
+  useEffect(() => {
```
Comment (Hyperkid123) [inline src/smart-components/order-process/order-processes.js:114]: you should also highlight the chrome navigation when landing in this page directly.
`insights.chrome.appNavClick({ id: 'page-id', secondaryNav: true });`

File: src/smart-components/order-process/order-processes.js [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+      </TopToolbar>
+      <TableToolbarView
+        sortBy={sortBy}
+        onSort={onSort}
+        data={data}
+        createRows={createRows}
+        columns={columns(intl)}
+        fetchData={updateOrderProcesses}
+        titlePlural={intl.formatMessage(orderProcessesMessages.orderProcesses)}
+        titleSingular={intl.formatMessage(orderProcessesMessages.orderProcess)}
```
Comment (Hyperkid123) [inline src/smart-components/order-process/order-processes.js:152]: This should be handled by the react int and some variable.

Comment (lgalis) [reply src/smart-components/order-process/order-processes.js:154]: the singular and plural are used in different places.

File: src/presentational-components/shared/table-toolbar-view.js (hunk 8 above)
Comment (lgalis) [reply src/presentational-components/shared/table-toolbar-view.js]: the singular and plural are used in different places.

File: src/presentational-components/shared/table-toolbar-view.js [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+              sortBy={sortBy}
+              onSort={onSort}
+            >
+              <TableHeader />
+              <TableBody />
+            </Table>
+          )}
+          {pagination.count > 0 && (
+            <PrimaryToolbar
+              className="pf-u-pl-lg pf-u-pr-lg ins__approval__primary_toolbar"
```
Comment (Hyperkid123) [inline src/presentational-components/shared/table-toolbar-view.js:122]: there are some approval classnames

File: src/presentational-components/shared/table-toolbar-view.js [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+    perPage: pagination.limit,
+    onSetPage: (_e, page) =>
+      fetchData({ ...pagination, offset: getNewPage(page, pagination.limit) }),
+    onPerPageSelect: (_e, size) => fetchData({ ...pagination, limit: size }),
+    isDisabled: isLoading
+  };
+
+  const renderToolbar = () => (
+    <PrimaryToolbar
+      className="pf-u-p-lg ins__approval__primary_toolbar"
```
Comment (Hyperkid123) [inline src/presentational-components/shared/table-toolbar-view.js:53]: This is probably from approval UI

**Pull request merged**